core/src/main/java/org/infinispan/util/logging/Log.java

```diff
@@ -1396,9 +1396,12 @@ void preparedTxAlreadyExists(RecoveryAwareTransaction previous,
    TimeoutException coordinatorTimeoutWaitingForView(int expectedViewId, int currentViewId,
                                                      Object clusterManagerStatus);
 
-   @LogMessage(level = WARN)
-   @Message(value = "No indexable classes were defined for this indexed cache; switching to autodetection (support for autodetection will be removed in Infinispan 11).", id = 403)
-   void noIndexableClassesDefined();
+   @Message(value = "No indexable classes were defined for this indexed cache. The configuration must contain " +
+         "classes or protobuf message types annotated with '@Indexed'", id = 403)
+   CacheConfigurationException noIndexableClassesDefined();
+
+   @Message(value = "Cache configuration must not declare indexed entities if it is not indexed", id = 404)
```

Comment:
I know 404 is probably the most important number in the Universe (404 > 42), but we should not reuse message ids :)